components/tyk-enterprise-portal/templates/statefulset-enterprise-portal.yaml

```diff
@@ -0,0 +1,211 @@
+      release: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app: {{ include "tyk-enterprise-portal.fullname" . }}
+        release: {{ .Release.Name }}
+      annotations:
+        checksum/secrets: "{{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}"
+    spec:
+{{- if .Values.nodeSelector }}
+      nodeSelector:
+{{ toYaml .Values.nodeSelector | indent 8 }}
+{{- end }}
+{{- if .Values.tolerations }}
+      tolerations:
+{{ toYaml .Values.tolerations | indent 8 }}
+{{- end }}
+{{- if .Values.affinity }}
+      affinity:
+{{ toYaml .Values.affinity | indent 8 }}
+{{- end }}
+      containers:
+      - image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        name: enterprise-portal-{{ .Chart.Name }}
+        securityContext:
+          runAsNonRoot: true
+          allowPrivilegeEscalation: false
+          privileged: false
+          readOnlyRootFilesystem: true
+          capabilities:
+            drop:
+              - all
+        env:
+          - name: PORTAL_LICENSEKEY
+            valueFrom:
+              secretKeyRef:
+                {{ if .Values.secrets.useSecretName }}
+                name: {{ .Values.secrets.useSecretName }}
+                {{ else }}
+                name: secrets-{{ include "tyk-enterprise-portal.fullname" . }}
+                {{ end }}
+                key: EnterprisePortalLicense
+          - name: PORTAL_HOST_PORT
+            value: "{{ .Values.containerPort }}"
+          - name: ADMIN_EMAIL
+            value: "{{ .Values.global.adminUser.email }}"
+          - name: ADMIN_PASSWORD
+            valueFrom:
+              secretKeyRef:
+                {{ if .Values.secrets.useSecretName }}
+                name: {{ .Values.secrets.useSecretName }}
+                {{ else }}
+                name: secrets-{{ include "tyk-enterprise-portal.fullname" . }}
+                {{ end }}
+                key: EnterprisePortalAdminPassword
+          - name: PORTAL_DATABASE_DIALECT
+            value: "{{ .Values.storage.database.type }}"
+          - name: PORTAL_DATABASE_CONNECTIONSTRING
+            valueFrom:
+              secretKeyRef:
+                {{ if .Values.secrets.useSecretName }}
+                name: {{ .Values.secrets.useSecretName }}
+                {{ else }}
+                name: secrets-{{ include "tyk-enterprise-portal.fullname" . }}
+                {{ end }}
+                key: EnterprisePortalStorgeConnectionString
+          - name: PORTAL_THEMING_THEME
+            value: "default"
+          - name: PORTAL_THEMING_PATH
+            value: "./themes"
+          - name: PROVIDER_NAME
+            value: "tyk"
+          - name: TYK_URL
+            value: "http{{ if .Values.global.tls.dashboard }}s{{ end }}://dashboard-svc-{{ .Release.Name }}-tyk-dashboard.{{ .Release.Namespace }}.svc:{{ .Values.global.servicePorts.dashboard }}"
```

Comment:
what happens if the dev port is installed as a subchart? I mean without installing it via an umbrella chart? Will the `tyk-dashboard` suffix appear?

Reply:
makes sense. let's make it configurable as you suggested. As a default value, I think we can use `"http{{ if .Values.global.tls.dashboard }}s{{ end }}://dashboard-svc-{{ .Release.Name }}-tyk-dashboard.{{ .Release.Namespace }}.svc:{{ .Values.global.servicePorts.dashboard }}"`

If users installs component developer portal, they can change it

Reply:
Added required change